atoma-state/src/handlers.rs

```diff
@@ -1827,4 +1902,176 @@ pub mod remote_attestation_verification {
+    ///
+    /// # Arguments
+    ///
+    /// * `state_manager` - A reference to the `AtomaStateManager` that provides access to the state database
+    /// * `nvswitch_evidence_data` - A reference to the NVSwitch evidence data
+    /// * `new_public_key` - The new public key
+    /// * `device_type` - The device type
+    ///
+    /// # Returns
+    ///
+    /// * `Result<()>` - Ok(()) if the attestation is successful, or an error if the operation failed
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    /// * The GPU evidence data cannot be decoded
+    /// * The NVSwitch evidence data cannot be decoded
+    #[instrument(
+        level = "info", 
+        skip_all,
+        fields(
+            device_type = device_type,
+            new_public_key = hex::encode(new_public_key),
+        ),
+        err,
+    )]
+    pub async fn attest_nvidia_nvswitch_evidence_list(
+        state_manager: &AtomaStateManager,
+        nvswitch_evidence_data: &[NvSwitchEvidence],
+        new_public_key: &[u8],
+        device_type: u16,
+    ) -> Result<()> {
+        let contract_nonce = state_manager
+            .state
+            .get_contract_key_rotation_nonce()
+            .await
+            .map_err(|_| AtomaStateRemoteAttestationError::FailedToRetrieveContractNonce)?;
+        let should_be_nonce = blake3::hash(
+            &[
+                &contract_nonce.to_le_bytes()[..],
+                new_public_key,
+                &device_type.to_le_bytes()[..],
+            ]
+            .concat(),
+        );
+        let should_be_nonce_hex = hex::encode(should_be_nonce.as_bytes());
+        let (result, _) = verify_nvswitch_attestation(
+            nvswitch_evidence_data,
+            &should_be_nonce_hex,
+            AttestRemoteOptions::default(),
+        ).await.map_err(|_| {
+            tracing::error!(
+                target = "atoma-state-handlers",
+                event = "attest-nvidia-evidence-list",
+                "NVSwitch attestation verification failed for device type: {device_type} and public key: {}",
+                hex::encode(new_public_key),
+            );
+            AtomaStateRemoteAttestationError::FailedToAttestRemote(
+                AttestError::RemoteAttestationFailed,
+            )
+        })?;
+        if result {
+            tracing::info!(
+                target = "atoma-state-handlers",
+                event = "attest-nvidia-evidence-list",
+                "Attestation successful for device type: {device_type} and public key: {}",
+                hex::encode(new_public_key),
+            );
+        } else {
+            tracing::error!(
+                target = "atoma-state-handlers",
+                event = "attest-nvidia-evidence-list",
+                "Attestation failed for device type: {device_type} and public key: {}",
+                hex::encode(new_public_key),
+            );
+        }
+
+        Ok(())
```

Comment:
The function returns `Ok(())` even when `result` is **false**, which contradicts the function's purpose of attestation. It logs an error but still returns success. Is this desired?